client/components/domains/registrant-extra-info/ca-form.jsx

```diff
@@ -160,24 +165,19 @@ class RegistrantExtraInfoCaForm extends React.PureComponent {
 	}
 
 	render() {
-		const { getFieldProps, translate } = this.props;
+		const { translate } = this.props;
 		const { legalType, ciraAgreementAccepted } = {
 			...defaultValues,
 			...this.props.contactDetailsExtra,
 		};
 
-		const rawFieldProps = getFieldProps( 'organization', true );
-
 		// We have to validate the organization name for the CCO legal
 		// type to avoid OpenSRS rejecting them and causing errors during
 		// payments
-		const organizationFieldProps = this.validateOrganizationIsNotEmpty( rawFieldProps );
-
-		const doesntNeedOrganizationField = legalType !== 'CCO';
-		const organizationFieldIsValid =
-			doesntNeedOrganizationField || ! organizationFieldProps.isError;
-
-		const formIsValid = ciraAgreementAccepted && organizationFieldIsValid;
+		const organizationFieldProps = this.addRequiredOrganizationValidation(
+			this.props.getFieldProps( 'organization', true )
```

Comment:
Aside from the imminent changes introduced by #19889, I'm not sure it's a good idea to pass state controllers between forms. The most obvious case I can think of is when someone decides to change the properties of the originating field in `domain-details-form` for whatever reason. I know it's unlikely for now, but I'm just being overly cautious.

Also `ca-form`, just like `fr-form`, is a self-contained form under `<ExtraInfoForm />` with idiosyncratic validation anyway. If we have to embed further regulatory checks later, managing the form's state internally might make it easier to test in isolation.

If we need the same validation across forms, we could abstract the validator function itself, or pass it as a prop. And we can already grab/update contact details in the store. 

What do you think?